packages/near-api-js/test/wallet-account.test.js

```diff
@@ -154,12 +155,63 @@ it('can complete sign in', async () => {
     await walletConnection._completeSignInWithAccessKey();
 
     expect(await keyStore.getKey('networkId', 'near.account')).toEqual(keyPair);
-    expect(localStorage.getItem('contractId_wallet_auth_key'));
+    expect(localStorage.getItem('contractId_wallet_auth_key')).toBeTruthy();
     expect(history.slice(1)).toEqual([
         [{}, 'documentTitle', 'http://example.com/location']
     ]);
 });
 
+it('can get the default appKeyPrefix', async () => {
+    nearFakeWithoutContract = {
+        config: {
+            networkId: 'networkId',
+            walletUrl: 'http://example.com/wallet',
+        },
+        connection: {
+            networkId: 'networkId',
+            signer: new nearApi.InMemorySigner(keyStore)
+        },
+        account() {
+            return {
+                state() { }
+            };
+        }
+    };
+
+    const keyPair = nearApi.KeyPair.fromRandom('ed25519');
+    global.window.location.href = `http://example.com/location?account_id=near.account&public_key=${keyPair.publicKey}`;
+    await keyStore.setKey('networkId', 'pending_key' + keyPair.publicKey, keyPair);
+
+    const newWalletConnection = new nearApi.WalletConnection(nearFakeWithoutContract);
+    await newWalletConnection._completeSignInWithAccessKey();
+
+    expect(localStorage.getItem('default_wallet_auth_key')).toBeTruthy();
+});
+
+it('can get the near-app appKeyPrefix', async () => {
+    const keyPair = nearApi.KeyPair.fromRandom('ed25519');
+    global.window.location.href = `http://example.com/location?account_id=near.account&public_key=${keyPair.publicKey}`;
+    await keyStore.setKey('networkId', 'pending_key' + keyPair.publicKey, keyPair);
+
+    const newWalletConnection = new nearApi.WalletConnection(nearFake, 'near-app');
+    await newWalletConnection._completeSignInWithAccessKey();
+
+    expect(localStorage.getItem('near-app_wallet_auth_key')).toBeTruthy();
+});
+
+it('can migrate from an existing appKeyPrefix successfully', async () => {
+    const keyPair = nearApi.KeyPair.fromRandom('ed25519');
+    global.window.location.href = `http://example.com/location?account_id=near.account&public_key=${keyPair.publicKey}`;
+    await keyStore.setKey('networkId', 'pending_key' + keyPair.publicKey, keyPair);
+
+    localStorage.setItem('undefined_wallet_auth_key');
+
+    const newWalletConnection = new nearApi.WalletConnection(nearFake, 'near-app');
+    await newWalletConnection._completeSignInWithAccessKey();
+
+    expect(localStorage.getItem('near-app_wallet_auth_key')).toBeTruthy();
+});
```

Comment:
This test does not execute migration since `authDataKey` is `near-app_wallet_auth_key` when the `WalletConnection` is constructed which is the same value as `DEPRECATED_authDataKey`. The following block would test it. Please check if there are other cases we are missing:
```suggestion
describe('can migrate from an existing appKeyPrefix successfully when not defined', () => {
    it('migrates to default_wallet_auth_key when no contract in near', async () => {
        const keyPair = nearApi.KeyPair.fromRandom('ed25519');
        await keyStore.setKey('networkId', 'pending_key' + keyPair.publicKey, keyPair);

        localStorage.setItem('undefined_wallet_auth_key', JSON.stringify({accountId: 'near.account', allKeys: [keyPair.toString()]}));

        new nearApi.WalletConnection(nearFakeWithoutContract);

        expect(localStorage.getItem('undefined_wallet_auth_key')).toBeNull();
        expect(localStorage.getItem('default_wallet_auth_key')).toEqual(JSON.stringify({accountId: 'near.account', allKeys: [keyPair.toString()]}));
    });

    it('migrates to contractId_wallet_auth_key when contract exists in near', async () => {
        const keyPair = nearApi.KeyPair.fromRandom('ed25519');
        await keyStore.setKey('networkId', 'pending_key' + keyPair.publicKey, keyPair);

        localStorage.setItem('undefined_wallet_auth_key', JSON.stringify({accountId: 'near.account', allKeys: [keyPair.toString()]}));

        new nearApi.WalletConnection(nearFake);

        expect(localStorage.getItem('undefined_wallet_auth_key')).toBeNull();
        expect(localStorage.getItem('contractId_wallet_auth_key')).toEqual(JSON.stringify({accountId: 'near.account', allKeys: [keyPair.toString()]}));
    });
});
```